bindings/azure/eventgrid/eventgrid.go

```diff
@@ -0,0 +1,186 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package eventgrid
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/eventgrid/mgmt/2019-06-01/eventgrid"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/dapr/pkg/logger"
+	"github.com/valyala/fasthttp"
+)
+
+// AzureEventGrid allows sending/receiving Azure Event Grid events
+type AzureEventGrid struct {
+	metadata *azureEventGridMetadata
+	logger   logger.Logger
+}
+
+type azureEventGridMetadata struct {
+	TenantID                  string `json:"tenantId"`
+	SubscriptionID            string `json:"subscriptionId"`
+	ResourceGroupName         string `json:"resourceGroupName"`
+	TopicName                 string `json:"topicName"`
+	EventGridSubscriptionName string `json:"eventGridSubscriptionName"`
+	SubscriberEndpoint        string `json:"subscriberEndpoint"`
+	TopicEndpoint             string `json:"topicEndpoint"`
+	ClientID                  string `json:"clientId"`
+	ClientSecret              string `json:"clientSecret"`
+}
+
+// NewAzureEventGrid returns a new Azure Event Grid instance
+func NewAzureEventGrid(logger logger.Logger) *AzureEventGrid {
+	logger.Debug("NewAzureEventGrid() called...")
+	return &AzureEventGrid{logger: logger}
+}
+
+// Init performs metadata init
+func (a *AzureEventGrid) Init(metadata bindings.Metadata) error {
+	a.logger.Debugf("Parsing Event Grid metadata(%s)...", metadata.Name)
+
+	m, err := a.parseMetadata(metadata)
+	if err != nil {
+		return err
+	}
+	a.metadata = m
+
+	a.logger.Debug("Metadata parsed successfully.")
+
+	return nil
+}
+
+func (a *AzureEventGrid) Read(handler func(*bindings.ReadResponse) error) error {
+	a.logger.Debug("Read() called...")
+
+	err := a.createSubscription()
+	if err != nil {
+		return err
+	}
+
+	m := func(ctx *fasthttp.RequestCtx) {
+		switch string(ctx.Path()) {
+		case "/api/events":
+			if string(ctx.Method()) == "OPTIONS" {
+				ctx.Response.Header.Add("WebHook-Allowed-Origin", string(ctx.Request.Header.Peek("WebHook-Request-Origin")))
+				ctx.Response.Header.Add("WebHook-Allowed-Rate", "*")
+				ctx.Response.Header.SetStatusCode(fasthttp.StatusOK)
+				_, err = ctx.Response.BodyWriter().Write([]byte(""))
+				if err != nil {
+					a.logger.Error(err.Error())
+				}
+			} else if string(ctx.Method()) == "POST" {
+				bodyBytes := ctx.PostBody()
+
+				a.logger.Debug(string(bodyBytes))
+				err = handler(&bindings.ReadResponse{
+					Data: bodyBytes,
+				})
+				if err != nil {
+					a.logger.Error(err.Error())
+				}
+			}
+		}
+	}
+
+	fasthttp.ListenAndServe(":8080", m)
```

Comment:
Should the port not be configurable?

Reply:
Good catch, @RicardoNiepel! Meant to add this and missed it. Just added.